File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.js
function solution(s) {
    let answer = [0, 0];
    
    while (s !== "1") {
        
        let tmp = "";
        let cnt = 0;
        for (let i = 0; i < s.length; i++) {
            if (s[i] === "1") {
                tmp += "1";
            }
            else {
                cnt++;
            }
        }
        
        s = tmp.length.toString(2);
        answer[0]++ 
        answer[1] += cnt;
    }
    
    
    return answer;
}

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    answer = []
    
    cnt = 0
    zero = 0
    
    while s != "1":
        tmp = ""
        for n in s:
            if n == "1":
                tmp += "1"
            else:
                zero += 1
        
        s = bin(len(tmp))[2:]
        cnt += 1
    
    answer = [cnt, zero]
    
    return answer